docs/design/kserve-integration-proposal.md

```diff
@@ -0,0 +1,1857 @@
+└────────────────────────────────────────────────────────────────┘
+                               │
+                               ▼
+┌────────────────────────────────────────────────────────────────┐
+│                    Worker Nodes (DaemonSet)                    │
+│                                                                │
+│  LocalModelNode Agent (Enhanced):                              │
+│  - Detect GPU hardware (nvidia-smi/rocm-smi)       # NEW      │
+│  - Download model weights (existing, unchanged)                │
+│  - Download kernel cache OCI images                # NEW      │
+│  - Verify signatures with cosign                   # NEW      │
+│  - Validate GPU compatibility with MCV             # NEW      │
+│  - Extract caches to /mnt/models/kernel-caches/    # NEW      │
+│  - Update status with GPU info and cache status    # NEW      │
+│                                                                │
+│  PersistentVolume (Node-Local):                                │
+│  /mnt/models/                                                  │
+│  ├── models/                    # Existing                    │
+│  │   └── llama-7b/              # Model weights               │
+│  └── kernel-caches/             # NEW                         │
+│      └── llama-7b/              # GPU kernels                 │
+└────────────────────────────────────────────────────────────────┘
+                               │
+                               ▼
+┌────────────────────────────────────────────────────────────────┐
+│                    InferenceService Pod                        │
+│                                                                │
+│  Predictor Container (vLLM/Triton/PyTorch):                    │
+│    - Model PVC mounted at /mnt/models                         │
+│    - Reads weights from /mnt/models/models/llama-7b           │
+│    - Reads kernels from /mnt/models/kernel-caches/llama-7b    │
+│    - Env: VLLM_KERNEL_CACHE=/mnt/models/kernel-caches/llama-7b│
+└────────────────────────────────────────────────────────────────┘
+```
+
+#### How the Feature Works
+
+**1. User Creates LocalModelCache with Kernel Cache:**
+
+Users extend their existing LocalModelCache resources with an optional `kernelCache` field:
+
+```yaml
+apiVersion: serving.kserve.io/v1alpha1
+kind: LocalModelCache
+metadata:
+  name: llama-7b
+spec:
+  # Existing fields (no changes)
+  sourceModelUri: s3://models/llama-7b
+  modelSize: 13Gi
+  nodeGroups:
+    - gpu-node-group
+
+  # NEW: Optional kernel cache configuration
+  kernelCache:
+    image: quay.io/myorg/llama-7b-vllm-kernels:v1
+    cacheSize: 500Mi
+    framework: vllm
+    gpuRequirements:
+      gpuType: A100
+      computeCapability: "8.0"
+    signaturePolicy:
+      require: true
+      issuer: "https://github.com/login/oauth"
+      subject: "https://github.com/myorg/ml-kernels/.github/workflows/build.yml@refs/heads/main"
+```
+
```

Comment:
The webhook should kick in to validate the image signature and translate the tag to the image sha